crates/ty_python_semantic/src/types/class_base.rs

```diff
@@ -123,7 +123,32 @@ impl<'db> ClassBase<'db> {
                     Some(valid_element)
                 }
             }
-            Type::Union(_) => None, // TODO -- forces consideration of multiple possible MROs?
+            Type::Union(union) => {
+                // We do not support full unions of MROs (yet). Until we do,
+                // support the cases where one of the types in the union is
+                // a dynamic type such as `Any` or `Unknown`, and all other
+                // types *would be* valid class bases. In this case, we can
+                // "fold" the other potential bases into the dynamic type,
+                // and return `Any`/`Unknown` as the class base to prevent
+                // invalid-base diagnostics and further downstream errors.
+                let Some(Type::Dynamic(dynamic)) = union
+                    .elements(db)
+                    .iter()
+                    .find(|elem| matches!(elem, Type::Dynamic(_)))
+                else {
+                    return None;
+                };
+
+                if union
+                    .elements(db)
+                    .iter()
+                    .all(|elem| ClassBase::try_from_type(db, *elem).is_some())
+                {
+                    Some(ClassBase::Dynamic(*dynamic))
+                } else {
+                    None
+                }
+            }
```

Comment:
I think we should still emit an error if it's a union with three or more elements. For a union `A | B | Unknown`, the type will still be a union type whatever the `Unknown` element materializes to. That means that whatever the `Unknown` element materialized to, we wouldn't be able to infer a correct MRO with our current capabilities, and I think that's something that most developers would want their type checker to warn them about, since it severely limits our ability to make accurate inferences about their code.

Test case:

```py
from typing import Any

class Foo: ...
class Bar: ...

def f(a: bool, b: bool, c: Any):
    if a:
        x = Foo
    elif b:
        x = Bar
    else:
        x = c

    class Baz(x): ...
```

Reply:
> For a union `A | B | Unknown`, the type will still be a union type whatever the `Unknown` element materializes to.

Not if it materializes to `object`?

Reply:
It could resolve to the type "instance of `object`", yes, but that would still be an invalid class base. In order for it to be a valid class base, the union would have to resolve to "the class object `object`". I don't think that's possible?

Reply:
Ah, that's a good point; we have to inherit from a class literal type, which is a singleton type, so there's no way it could collapse a union to a valid base. So I think you're right that we could error on a union containing Any/Unknown if it has two other elements.